SensorCode/SensorState.py

```diff
@@ -34,12 +43,72 @@ def set_lidar(self, data):
             self.lidar[i] = j
 
     def get_lidar(self):
-        ans= []
-        for i, j in enumerate(self.lidar):
-            if j==None:
-                continue
-            ans.append((i,j))
-        return ans
+        vis_map = {} #a dictionary associating angles with object distance
+        vis_angles = [False] * 360 #List of visited angles with a margin of +-2
+        count = 0
+        while count < 360:
+            # list_tup = LIDAR_API.get_LIDAR_tuples()
+            list_tup = TEST_API.get_array("LDR")
+            print(list_tup)
+            for ang, dist in list_tup:
+                print(ang)
+                if ang not in vis_map:
+                    for index_offset in [-2, -1, 0, 1, 2]:
+                        #angles with +-2 of read-in angle are also treated as visited
+                        print("test: " + str(ang-index_offset) + " ")
+                        if 0 <= ang-index_offset < 360 and not vis_angles[ang-index_offset]:
+                            vis_angles[ang-index_offset] = True
+                            count += 1
+                vis_map[ang] = dist
+        # self.lidar = list(vis_map.items())
+        return list(vis_map.items()) #or just do line above and refactor servergui?
+
+        # ans= []
+        # for i, j in enumerate(self.lidar):
+        #     if j==None:
+        #         continue
+        #     ans.append((i,j))
+        # return ans
+
+
+    def get_terabee(self):
+        """
+
+        :return: 3 lists of tuples representing (angle, distance) for bottom, mid, top terabee sensors respectively
+        """
+        returner1 = [] # bottom terabee list of tuples
+        returner2 = [] # mid terabee list of tuples
+        returner3 = [] # top terabe list of tuples
+
+        counter = 0
+        for distance in self.terabee_bot:
```

Comment:
If you want to go the extra mile and do it in two lines instead of 4

for i, distance in zip(range(len(self.terabee_bot)), self.terabee_bot):
    returner1.append((self.terabee_bot_ang[i], distance))